discovery/kubernetes/kubernetes.go

```diff
@@ -724,6 +724,19 @@ func (d *Discovery) newEndpointsByNodeInformer(plw *cache.ListWatch) cache.Share
 
 func (d *Discovery) newEndpointSlicesByNodeInformer(plw *cache.ListWatch, object runtime.Object) cache.SharedIndexInformer {
 	indexers := make(map[string]cache.IndexFunc)
+	indexers[serviceIndex] = func(obj interface{}) ([]string, error) {
+		var services []string
+		switch e := obj.(type) {
+		case *disv1.EndpointSlice:
+			if serviceName, exists := e.ObjectMeta.Labels[disv1.LabelServiceName]; exists {
+				return []string{serviceName}, nil
```

Comment:
the service name in the label doesn't seem to be namespaced.
we should use `namespacedName()` to ensure uniqueness (two services in two different namespaces can have the same name)

if you could add add a `TestEndpointSliceDiscoveryWithUnrelatedServiceUpdate` where we update an unrelated service with the same name, in another namespace and check the no-op, it'd be a great regression test.

Reply:
`TestEndpointSliceDiscoveryWithUnrelatedServiceUpdate` is added at the new commit.